metadata-ingestion/tests/unit/sdk_v2/test_search_client.py

```diff
@@ -240,6 +240,18 @@ def test_field_discriminator() -> None:
         == _CustomCondition._field_discriminator()
     )
 
+    class _BadFilter(_BaseFilter):
+        field1: str
+        field2: str
+
+    with pytest.raises(
+        ValueError,
+        match=re.escape(
+            "Found multiple fields that could be the discriminator for this filter: ['field1', 'field2']"
+        ),
+    ):
+        _BadFilter._field_discriminator()
+
```

Comment:
The failing test is fine.

There may be another test covering the happy path. If I understand correctly, that would be the class overriding the `_field_discriminator` method.

Reply:
yup - the custom filter already tests the happy path